src/libraries/System.Private.CoreLib/src/System/Collections/HashHelpers.cs

```diff
@@ -15,6 +15,9 @@ internal static partial class HashHelpers
 
         public const int HashPrime = 101;
 
+        // This field is shared between Dictionary and HashSet for initially empty collections and shouldn't be written to
```

Comment:
This file is used outside CoreLib. These two new static fields should be under `#if SYSTEM_PRIVATE_CORELIB` so that the other uses do not pay for them.

Reply:
Done.